pkg/driver/controller_test.go

```diff
@@ -429,3 +429,204 @@ func TestPickAvailabilityZone(t *testing.T) {
 		})
 	}
 }
+
+func TestControllerPublishVolume(t *testing.T) {
+	fakeCloud := NewFakeCloudProvider()
```

Comment:
Could you create a `mock_cloud` using gomock? With that, we could truly unit test controller service without depending on the fake cloud implementation. 

Here is something similar I did for another driver: https://github.com/aws/aws-fsx-csi-driver/blob/master/pkg/driver/controller_test.go#L224

Reply:
That's a good idea but maybe we can do it in a follow up PR and introduce it in all the required tests. wdyt?

Reply:
Fair enough.

Reply:
#220